### PR TITLE
Fix coordinate messup when converting H3 entities to points. Fixes #58

### LINKED ISSUE #58: cells_to_wkb_points returning longitude for both coordinates
Hi! I've been migrating a project to `h3ronpy` but I've been running into an issue returning the centroids of hexagons. It seems that the coordinates returned have exactly the same value:

```python
import h3ronpy
from shapely.geometry import Point
from h3ronpy.arrow.vector import geometry_to_cells, cells_to_wkb_points
import pyarrow as pa
from shapely import wkb

# Step 1: Create a point (latitude and longitude)
lat, lon = 37.7749, -122.4194  # Example coordinates (San Francisco)
point = Point(lon, lat)  # shapely expects (longitude, latitude)

# Step 2: Convert the point to an H3 cell (resolution 9 for example)
resolution = 9
h3_cells = geometry_to_cells(point, resolution)

# Step 3: Convert the H3 cell back to WKB points
wkb_points = cells_to_wkb_points(h3_cells)

# Step 4: Decode the WKB point to a Shapely geometry
for wkb_point in wkb_points:
    if isinstance(wkb_point, pa.Scalar):  # Ensure it's a pyarrow Scalar
        shapely_point = wkb.loads(wkb_point.as_buffer().to_pybytes())
        print(f"Shapely Geometry: {shapely_point}")
    else:
        print(f"Unexpected Type: {type(wkb_point)}")
```

Returns:
```
Shapely Geometry: POINT (-122.41827103692466 -122.41827103692466)
```

I'd be happy to help with the fix!

### PR DESCRIPTION
**Pull request by nmandery**

### CODE REVIEW

File: h3ronpy/pyproject.toml [hunk 1]
```diff
@@ -37,6 +37,7 @@ polars = [
 ]
 pandas = [
     "geopandas>=0.10",
+    "fiona>=1.9"
```
Comment (kylebarron) [inline h3ronpy/pyproject.toml:40]: FYI https://github.com/geopandas/pyogrio is vectorized and much, much faster than fiona. It's the default for GeoPandas >= 1.0 but can be used opt-in before that as well.

File: h3ronpy/tests/arrow/test_vector.py [hunk 2]
```diff
@@ -20,3 +21,25 @@ def test_geometry_to_cells_central_park():
...
+    resolution = 9
+    h3_cells = geometry_to_cells(point, resolution)
+
+    # Step 3: Convert the H3 cell back to WKB points
+    wkb_points = cells_to_wkb_points(h3_cells)
+
+    assert len(wkb_points) == 1
+
+    # Step 4: Decode the WKB point to a Shapely geometry
+    for wkb_point in wkb_points:
```
Comment (kylebarron) [inline h3ronpy/tests/arrow/test_vector.py:41]: I know this is just a test file and speed doesn't matter as much (and maybe this is a list of one?) but `shapely.from_wkb` is vectorized and much faster than a loop and `shapely.wkb.loads`.

**Review by kylebarron — APPROVED** [review]

File: h3ronpy/pyproject.toml (hunk 1 above)
Comment (kylebarron) [reply h3ronpy/pyproject.toml:40]: Also, the failing test is that you either need to upgrade geopandas or require fiona `<1.10` https://stackoverflow.com/questions/78949093/how-to-resolve-attributeerror-module-fiona-has-no-attribute-path

Comment (nmandery) [reply h3ronpy/pyproject.toml]: Thanks you. I am mostly trying to get this to build and to release a patch version with as little breaks as possible. So I think I will go now with the fiona downgrade. After that its is time to work on a new minor-release with the other issues, a few dependency upgrades and the requirement of python >= 3.9

File: h3ronpy/tests/arrow/test_vector.py (hunk 2 above)
Comment (nmandery) [reply h3ronpy/tests/arrow/test_vector.py:41]: I did not know that. Here I was lazy and just took the code from the original issue and copied it into a unittest ;)

**Pull request merged**